### DISCUSSION TITLE
Bottom screen

### DISCUSSION THREAD

**Discussion by MidyGamy** [profi200/open_agb_firm#214] [Ideas]
I didn't find a way to output the game on the bottom screen... Would it be possible to add an option for it to the configuration file?

**Comment by profi200** [comment]
I don't have a library like OpenGL available. Everything is hardcoded and it's a mess to support more configurations. On top of that the scaling ratio for the bottom LCD is even worse at ~1.333333333333333333333. Top is just 1.5 which is not great but easier to archive.

**Reply by MidyGamy** [reply]
Oh man... I see...

**Reply by wongban** [reply]
Top screen is a 3D screen, There will be horizontal lines when used. The bottom screen does not have this problem. The effect may be better. [oaf-boost](https://oafc.anod.cc/) does this, maybe it can be used as a reference?

**Reply by Masamune3210** [reply]
There will be horizontal lines regardless of which screen is used. There were horizontal lines on the GBA screen as well. Thats just how LCD's work.

**Reply by wongban** [reply]
The following two pictures are the 1:1 real-life effects of the upper and lower screens of the new 3ds xl. If you zoom in on the pictures, you can see a big difference.
![IMG_4620](https://github.com/user-attachments/assets/26bc0aa5-6d1a-4ebb-8bf1-6d91a0de1bfd)
![IMG_4619](https://github.com/user-attachments/assets/0d12b191-05af-4743-bc81-3be2b83b9a64)
![屏幕截图 2025-01-04 130118](https://github.com/user-attachments/assets/e3a82d99-6001-4391-97b2-14c2bfb1d05a)
Maybe the GBA is more like a dot matrix effect rather than a scan line.

[refer to:](https://www.reddit.com/media?url=https%3A%2F%2Fpreview.redd.it%2Fn2jcdynkne461.png%3Fwidth%3D1913%26format%3Dpng%26auto%3Dwebp%26s%3D48d5ee5c7410f60aa746acc66115d91236fdf07a)

So I think the lower screen can better restore the display effect of GBA.

**Reply by Masamune3210** [reply]
In the end, you have to get way closer to the screen than would be comfortable to play, so the point is kind of moot.

Is there a difference? Yes. 
Does it change that much between them? Not really.

**Reply by profi200** [reply]
Both LCDs are physically rotated and as a result the subpixels are rotated too. I found out via microscope shots that if red and blue subpixels are stacked on top of each other you get a CRT TV like effect meaning there are horizontal "black" lines. In reality there are basically none. There are vertical lines. If you rotate your 3DS 90° you will see the lines disappear and the LCD suddenly looks very good.
